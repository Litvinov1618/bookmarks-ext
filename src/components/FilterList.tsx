import React, { useState, useEffect } from "react";
import styled from "styled-components";

const Wrapper = styled.nav`
  width: 300px;
  margin-bottom: 10px;
  position: relative;

  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;

const Filter = styled.div`
  width: 90px;
  height: 100%;
  border: 1px solid #e95656;
  box-sizing: border-box;
  border-radius: 5px;
  position: relative;

  &:hover {
    cursor: pointer;
  }
`;

const FilterName = styled.div`
  font-style: normal;
  font-weight: normal;
  font-size: 18px;
  line-height: 25px;
  text-align: center;
`;

const FilterItem = styled.div`
  display: none;
  font-style: normal;
  font-weight: normal;
  font-size: 18px;
  line-height: 25px;
  text-align: center;
  overflow: hidden;
  text-overflow: ellipsis;
  ${Filter}:hover & {
    display: block;
  }
`;

interface FilterListProps {
  documents: { data: () => { tags: string } }[];
  query: React.Dispatch<
    React.SetStateAction<
      firebase.firestore.CollectionReference<firebase.firestore.DocumentData>
    >
  >;
  collection: any;
}

const FilterList: React.FC<FilterListProps> = ({
  documents,
  query,
  collection,
}) => {
  const [tags, setTags] = useState<string[]>(["Loading"]);
  useEffect(() => {
    if (documents.length !== 0) {
      const documentsTags: string[] = [];
      documents.map(
        (document) =>
          document.data().tags.indexOf("") === -1 &&
          documentsTags.push(document.data().tags)
      );
      setTags(documentsTags);
    }
  }, [documents]);
  const filters = {
    readTime: ["Small", "Medium", "High"],
    interest: ["Small", "Medium", "High"],
    tags,
  };

  const handleFilterItem = (
    event: React.MouseEvent<HTMLDivElement, MouseEvent>
  ) => {
    applyFilters(event.target as HTMLInputElement);
  };

  const applyFilters = (element: HTMLInputElement) => {
    const filterName = element.parentElement!.id;
    if (filterName === "tags") {
      query(
        collection.where(
          `${filterName}`,
          "array-contains",
          `${element.innerText}`
        )
      );
    } else
      query(
        collection.where(
          `${filterName}`,
          "==",
          `${element.innerText.toLowerCase()}`
        )
      );
  };

  return (
    <Wrapper>
      <Filter id="time">
        <FilterName>Read Time</FilterName>
        {filters.readTime.map((filterItem) => (
          <FilterItem key={filterItem} onClick={handleFilterItem}>
            {filterItem}
          </FilterItem>
        ))}
      </Filter>
      <Filter id="interest">
        <FilterName>Interest</FilterName>
        {filters.interest.map((filterItem) => (
          <FilterItem key={filterItem} onClick={handleFilterItem}>
            {filterItem}
          </FilterItem>
        ))}
      </Filter>
      <Filter id="tags">
        <FilterName>Tags</FilterName>
        {filters.tags.map((filterItem) => (
          <FilterItem key={filterItem} onClick={handleFilterItem}>
            {filterItem}
          </FilterItem>
        ))}
      </Filter>
    </Wrapper>
  );
};

export default FilterList;
