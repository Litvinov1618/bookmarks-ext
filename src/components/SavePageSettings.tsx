import React, { useState } from "react";
import styled from "styled-components";
import TimeIcon from "./TimeIcon";
import InterestIcon from "./InterestIcon";
import useFirestoreCollection from "../components/Firebase/useFirestoreCollection";

const Label = styled.label`
  margin-bottom: 10px;
  display: block;
`;

const TagsArea = styled.textarea`
  border: 1px solid #e95656;
  box-sizing: border-box;
  border-radius: 5px;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  resize: none;

  height: 30px;
  padding: 5px;
  width: 40%;
`;

const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  height: 90px;
`;

const Button = styled.button`
  width: 90px;
  height: 30px;
  margin-left: 28px;
  padding: 5px;

  border: 1px solid #e95656;
  box-sizing: border-box;
  border-radius: 5px;
  background-color: white;
  font-size: 18px;

  &:hover {
    cursor: pointer;
  }
`;

const InputWrapper = styled.div`
  width: 50%;
`;

const InputGroup = styled.div`
  display: flex;
`;

const IconWrapper = styled.div`
  margin-right: 10px;
  display: inline-block;
  position: relative

  &:hover {
    cursor: pointer;
  }
`;

interface SavePageProps {
  url: string;
  title: string;
}

const SavePageSettings: React.FC<SavePageProps> = ({ url, title }) => {
  const [time, setTime] = useState("");
  const handleTime = (event: {
    target: { value: React.SetStateAction<string> };
  }) => {
    setTime(event.target.value);
    console.log(event.target.value);
  };

  const [interest, setInterest] = useState("");
  const handleInterest = (event: {
    target: { value: React.SetStateAction<string> };
  }) => {
    setInterest(event.target.value);
  };

  const [tags, setTags] = useState("");
  const handleTags = (event: {
    target: { value: React.SetStateAction<string> };
  }) => {
    setTags(event.target.value);
  };

  const { add } = useFirestoreCollection("pages", false);
  const [sendBtnStatus, setSendBtnStatus] = useState(false);
  const sendBookmark = () => {
    add({
      interest,
      tags: tags.split(", "),
      time,
      title,
      url,
    });
    setSendBtnStatus(true);
  };

  return (
    <Wrapper>
      <InputGroup>
        <InputWrapper>
          <Label>Read Time:</Label>
          <IconWrapper>
            <TimeIcon status="small" />
          </IconWrapper>
          <IconWrapper>
            <TimeIcon status="medium" />
          </IconWrapper>
          <IconWrapper>
            <TimeIcon status="high" />
          </IconWrapper>
        </InputWrapper>
        <InputWrapper>
          <Label>Interest:</Label>
          <IconWrapper>
            <InterestIcon status="small" />
          </IconWrapper>
          <IconWrapper>
            <InterestIcon status="medium" />
          </IconWrapper>
          <IconWrapper>
            <InterestIcon status="high" />
          </IconWrapper>
        </InputWrapper>
      </InputGroup>
      <InputGroup>
        <TagsArea placeholder="Tags" onChange={handleTags} value={tags} />
        <Button disabled={sendBtnStatus} onClick={sendBookmark}>
          Send
        </Button>
      </InputGroup>
    </Wrapper>
  );
};

export default SavePageSettings;
