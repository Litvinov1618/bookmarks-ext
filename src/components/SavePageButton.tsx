import React, { useState } from "react";
import styled from "styled-components";
import SavePageSettings from "./SavePageSettings";

const Wrapper = styled.header`
  border: 2px solid #e95656;
  box-sizing: border-box;
  border-radius: 20px;

  min-height: 50px;
  width: 300px;
  margin: 4px 0 10px 0;
  display: flex;
  flex-direction: column;
`;

const MainContent = styled.div`
  padding: 10px;
`;

const Span = styled.span`
  font-style: normal;
  font-weight: normal;
  font-size: 24px;
  line-height: 28px;

  display: block;
  margin-bottom: 5px;
`;

declare const chrome: {
  tabs: {
    query: (
      arg0: { active: boolean; lastFocusedWindow: boolean },
      arg1: (tabs: any[]) => void
    ) => void;
  };
};

const SavePageButton: React.FC = () => {
  const [pageUrl, setPageUrl] = useState("");
  const [pageTitle, setPageTitle] = useState("");
  const savePage = () => {
    if (process.env.REACT_APP_IS_EXTENSION) {
      chrome.tabs.query(
        { active: true, lastFocusedWindow: true },
        (tabs: Array<any>) => {
          setPageUrl(tabs[0].url);
          setPageTitle(tabs[0].title);
        }
      );
    }
  };
  return (
    <Wrapper>
      <MainContent onClick={savePage}>
        <Span>Save this page</Span>
        <SavePageSettings url={pageUrl} title={pageTitle} />
      </MainContent>
    </Wrapper>
  );
};

export default SavePageButton;
