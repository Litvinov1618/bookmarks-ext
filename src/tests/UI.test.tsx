import React from "react";
import renderer from "react-test-renderer";
import SavePageSettings from "../components/SavePageSettings";
import SavePageButton from "../components/SavePageButton";
import FilterList from "../components/FilterList";

it("SavePageSettings component renders correctly", () => {
  const component = renderer.create(<SavePageSettings />).toJSON();
  expect(component).toMatchSnapshot();
});

it("SavePageButton component renders correctly", () => {
  const component = renderer.create(<SavePageButton />).toJSON();
  expect(component).toMatchSnapshot();
});

it("FilterList component renders correctly", () => {
  const component = renderer.create(<FilterList />).toJSON();
  expect(component).toMatchSnapshot();
});
