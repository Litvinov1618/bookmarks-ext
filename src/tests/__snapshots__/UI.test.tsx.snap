// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterList component renders correctly 1`] = `
<nav
  className="sc-fzpans knaLVU"
>
  <div
    className="sc-fzplWN gSpIEx"
  >
    <div
      className="sc-fznyAO jNSMMS"
    >
      Read Time
    </div>
    <div
      className="sc-fznKkj ceWnMI"
      onClick={[Function]}
    >
      Small
    </div>
    <div
      className="sc-fznKkj ceWnMI"
      onClick={[Function]}
    >
      Medium
    </div>
    <div
      className="sc-fznKkj ceWnMI"
      onClick={[Function]}
    >
      High
    </div>
  </div>
  <div
    className="sc-fzplWN gSpIEx"
  >
    <div
      className="sc-fznyAO jNSMMS"
    >
      Interest
    </div>
    <div
      className="sc-fznKkj ceWnMI"
      onClick={[Function]}
    >
      Small
    </div>
    <div
      className="sc-fznKkj ceWnMI"
      onClick={[Function]}
    >
      Medium
    </div>
    <div
      className="sc-fznKkj ceWnMI"
      onClick={[Function]}
    >
      High
    </div>
  </div>
  <div
    className="sc-fzplWN gSpIEx"
  >
    <div
      className="sc-fznyAO jNSMMS"
    >
      Tags
    </div>
    <div
      className="sc-fznKkj ceWnMI"
      onClick={[Function]}
    >
      Loading
    </div>
  </div>
</nav>
`;

exports[`SavePageButton component renders correctly 1`] = `
<header
  className="sc-AxmLO cWabqP"
>
  <div
    className="sc-fzozJi BoHKt"
    onClick={[Function]}
  >
    <span
      className="sc-fzoLsD lpjFOM"
    >
      Save this page
    </span>
    <svg>
      arrow_down.svg
    </svg>
  </div>
</header>
`;

exports[`SavePageSettings component renders correctly 1`] = `
<div
  className="sc-AxgMl gnKkkU"
>
  <div>
    <label
      className="sc-AxhCb llbGyu"
    >
      Read Time:
    </label>
    <div
      className="sc-AxjAm itEsx"
    >
      <div
        className="sc-AxirZ dVgATR"
      >
        <input
          checked={false}
          className="sc-AxiKw klInzN"
          name="time"
          onChange={[Function]}
          type="radio"
          value="small"
        />
        <svg>
          small_time.svg
        </svg>
      </div>
      <div
        className="sc-AxirZ dVgATR"
      >
        <input
          checked={false}
          className="sc-AxiKw klInzN"
          name="time"
          onChange={[Function]}
          type="radio"
          value="medium"
        />
        <svg>
          medium_time.svg
        </svg>
      </div>
      <div
        className="sc-AxirZ dVgATR"
      >
        <input
          checked={false}
          className="sc-AxiKw klInzN"
          name="time"
          onChange={[Function]}
          type="radio"
          value="high"
        />
        <svg>
          high_time.svg
        </svg>
      </div>
    </div>
  </div>
  <div>
    <label
      className="sc-AxhCb llbGyu"
    >
      Interest:
    </label>
    <div
      className="sc-AxjAm itEsx"
    >
      <div
        className="sc-AxirZ dVgATR"
      >
        <input
          checked={false}
          className="sc-AxiKw klInzN"
          name="interest"
          onChange={[Function]}
          type="radio"
          value="small"
        />
        <svg>
          small_interest.svg
        </svg>
      </div>
      <div
        className="sc-AxirZ dVgATR"
      >
        <input
          checked={false}
          className="sc-AxiKw klInzN"
          name="interest"
          onChange={[Function]}
          type="radio"
          value="medium"
        />
        <svg>
          medium_interest.svg
        </svg>
      </div>
      <div
        className="sc-AxirZ dVgATR"
      >
        <input
          checked={false}
          className="sc-AxiKw klInzN"
          name="interest"
          onChange={[Function]}
          type="radio"
          value="high"
        />
        <svg>
          high_interest.svg
        </svg>
      </div>
    </div>
  </div>
  <div>
    <label
      className="sc-AxhCb llbGyu"
    >
      Tags:
    </label>
    <textarea
      className="sc-AxhUy bKTPOf"
      onChange={[Function]}
      value=""
    />
  </div>
  <div
    className="sc-AxheI ByOdM"
  >
    <button
      className="sc-Axmtr jQdBXG"
      disabled={false}
      onClick={[Function]}
    >
      Send
    </button>
  </div>
</div>
`;
